components/fxa-client/src/http_client.rs

```diff
@@ -185,6 +193,29 @@ impl FxAClient for Client {
         Self::make_request(request)?.json().map_err(Into::into)
     }
 
+    fn authorized_code_using_session_token(
+        &self,
+        config: &Config,
+        client_id: &str,
+        session_token: &str,
+        scope: &str,
+        state: &str,
+    ) -> Result<OAuthAuthResponse> {
+        let parameters = json!({
+            "client_id": client_id,
+            "scope": scope,
+            "response_type": "code",
+            "state": state,
```

Comment:
Since FPN needs a refresh token, the caller is going to need to be able to pass `access_type=offline` as a parameter when calling this method.